### ISSUE TITLE
 How can I troubleshoot when my code stops running after displaying "engine/trainer"?

### ISSUE THREAD

**Issue by maoliangjie** [ultralytics/ultralytics#18198]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How can I troubleshoot when my code stops running after displaying "engine/trainer"?

here is my training code:

from ultralytics import YOLO

model = YOLO("yolo11n.pt")
train_results = model.train(
    data="coco8.yaml",  # path to dataset YAML
    epochs=100,  # number of training epochs
    imgsz=640,  # training image size
    device=['0','1','2','3'],  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
)

**Comment by Y-T-G** [comment]
Can you post the logs?

Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by maoliangjie** [comment]
> Can you post the logs?
> 
> Can you provide the output after running this command in terminal: `yolo checks`?

of course.

yolo checks
Ultralytics 8.3.49 🚀 Python-3.8.20 torch-2.4.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3090, 24253MiB)
Setup complete ✅ (256 CPUs, 377.5 GB RAM, 281.8/937.3 GB disk)

OS                  Linux-6.8.0-49-generic-x86_64-with-glibc2.17
Environment         Linux
Python              3.8.20
Install             pip
RAM                 377.52 GB
Disk                281.8/937.3 GB
CPU                 AMD EPYC 9554 64-Core Processor
CPU count           256
GPU                 NVIDIA GeForce RTX 3090, 24253MiB
GPU count           4
CUDA                12.1

numpy               ✅ 1.24.4>=1.23.0
numpy               ✅ 1.24.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 2.4.1>=1.8.0
torch               ✅ 2.4.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.19.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0

I have deployed a Conda-based YOLO environment on a Linux server. After running to the point where it displays 'engine/trainer', the code becomes unresponsive. However, when I deploy the same environment on a Windows PC, it runs normally.
The content of train.py that was run is as follows.
python train.py 
Ultralytics 8.3.49 🚀 Python-3.8.20 torch-2.4.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3090, 24253MiB)
                                                      CUDA:1 (NVIDIA GeForce RTX 3090, 24253MiB)
                                                      CUDA:2 (NVIDIA GeForce RTX 3090, 24253MiB)
                                                      CUDA:3 (NVIDIA GeForce RTX 3090, 24253MiB)
engine/trainer: task=detect, mode=train, model=yolo11n.pt, data=coco8.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=['0', '1', '2', '3'], workers=8, project=None, name=train4, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train4

**Comment by Y-T-G** [comment]
Does it work with single GPU?

**Comment by maoliangjie** [comment]
> Does it work with single GPU?

No, same result. I have debuged this problem. the process stops at the "model.py" 800 self.trainer = (trainer or self._smart_load("trainer"))(overrides=args, _callbacks=self.callbacks)
After the smart_load return the trainer class, the code becomes unresponsive.
I don't know how to solve this, so I test the code in Windows PC. Then it works.

**Comment by Y-T-G** [comment]
Can you try with `workers=0`?

**Comment by maoliangjie** [comment]
> Can you try with `workers=0`?

Yeah, I have tried this. still same result

**Comment by Y-T-G** [comment]
Maybe try reinstalling PyTorch

**Comment by maoliangjie** [comment]
> Maybe try reinstalling PyTorch

Yeah, I also tried the python from 3.8 - 3.11, and corresponding torch. It still has this problem, while in my windows PC, my code works in conda environment which contains python 3.8 and corresponding torch

**Comment by glenn-jocher** [comment]
It seems like the issue might be environment-specific. Since it works on Windows but not Linux, consider verifying your Linux setup for potential conflicts, such as mismatched CUDA or PyTorch versions. Ensure your CUDA driver matches your installed PyTorch version. You can check compatibility here: https://pytorch.org/get-started/previous-versions/. If the issue persists, please try running with `TORCH_DISTRIBUTED_DEBUG=INFO` set in your environment to gather more debugging insights.
